lib/assertions/directory_equal.js

```diff
@@ -0,0 +1,69 @@
+module.exports = function (chai, utils) {
+
+	var Assertion = chai.Assertion;
+	var flag = utils.flag;
+	var assert = chai.assert;
+
+	var readdir = require('readdir-enhanced');
+	var format = require('util').format;
+
+	//-------------------------------------------------------------------------------------------------------------
+
+	Assertion.overwriteMethod('equal', function (_super) {
+		return function dirEqual (expected, msg) {
+			if (flag(this, 'fs.isDirectory')) {
+				var deep = flag(this, 'deep') ? 'deep ' : '';
+
+				var obj = this._obj;
+				var preMsg = '';
+				if (msg) {
+					flag(this, 'message', msg);
+					preMsg = msg + ': ';
+				}
+
+				new chai.Assertion(expected, preMsg + 'expected-value').is.a('string');
+				new chai.Assertion(expected, preMsg + 'expected-value').to.be.a.path();
+				new chai.Assertion(expected, preMsg + 'expected-value').to.be.a.directory();
+
+				var actualContents = readdir.sync(obj, {deep: !!deep, sep: '/'}).sort();
+				var expectedContents = readdir.sync(expected, {deep: !!deep, sep: '/'}).sort();
+
+				var pass;
+				try {
+					new chai.Assertion(actualContents, msg).has.same.members(expectedContents, msg);
+					pass = true;
+				}
+				catch (e) {
+					pass = false;
```

Comment:
Same thing here as prior comment about re-throwing the error if it's not an AssertionError.